### PR TITLE
Add HISTORY and clean RELEASE

### PR DESCRIPTION
**Pull request by byrnHDF**
develop RELEASE.txt is everything that has happened to the code since the creation of the 1.16 release branch plus any new not to be in 1.16 code.

### CODE REVIEW

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -47,12 +47,6 @@ New Features
 
     Configuration:
     -------------
-    - Added signed Windows msi binary and signed Apple dmg binary files.
-
-      The release process now provides signed Windows and Apple installation
-      binaries in addition to the debian and rpm installation binaries. Also
-      these installer files are no longer compressed into packaged archives.
-
     - Added configuration option for internal threading/concurrency support:
```
Comment (mattjala) [inline release_docs/RELEASE.txt:50]: Should this be removed too?

File: release_docs/RELEASE.txt [hunk 2]
```diff
@@ -65,399 +59,9 @@ New Features
...
-    - Added new routines for interacting with error stacks: H5Epause_stack,
-      H5Eresume_stack, and H5Eis_paused.  These routines can be used to
-      indicate that errors from a call to an HDF5 routine should not be
-      pushed on to an error stack.  Primarily targeted toward 3rd-party
-      developers of Virtual File Drivirs (VFDs) and Virtual Object Layer (VOL)
-      connectors, these routines allow developers to perform "speculative"
-      operations (such as trying to open a file or object) without requiring
-      that the error stack be cleared after a speculative operation fails.
-
     - H5Pset_external() now uses HDoff_t, which is always a 64-bit type
```
Comment (mattjala) [inline release_docs/RELEASE.txt:65]: Same

File: release_docs/RELEASE.txt [hunk 3]
```diff
@@ -819,1310 +121,37 @@ New Features
...
-
-      The error was fixed by inserting a break statement at the end of the case 'L': sections.
-
-      Fixes GitHub issue #3056
+    -
 
 
     Configuration
     -------------
-    - Changed name of libhdf5hl_fortran installed by autotools to libhdf5_hl_fortran.  The
```
Comment (mattjala) [inline release_docs/RELEASE.txt:1896]: Same

File: release_docs/RELEASE.txt [hunk 4]
```diff
@@ -2170,22 +186,11 @@ Bug Fixes since HDF5-1.14.0 release
...
-
-      The dt_arith test program's test_particular_fp_integer sub-test tries
-      to ensure that the library correctly raises a datatype conversion
-      exception when converting a floating-point value to an integer overflows.
-      However, this test would run even when H5_WANT_DCONV_EXCEPTION isn't
-      defined, causing the test to fail due to the library not raising
-      datatype conversion exceptions. This has now been fixed by not running
-      the test when H5_WANT_DCONV_EXCEPTION is not defined.
-
     - Disabled running of MPI Atomicity tests for OpenMPI major versions < 5
```
Comment (mattjala) [inline release_docs/RELEASE.txt:194]: Same

File: release_docs/RELEASE.txt (hunk 1 above)
Comment (byrnHDF) [reply release_docs/RELEASE.txt:50]: develop and 1.16 are initially the same thing so the Release.txt file should be the same

Comment (byrnHDF) [reply release_docs/RELEASE.txt:50]: Therefore anything that did not go into 1.14 releases need to be in both 1.16 and develop.

**Review by lrknox — APPROVED** [review]

**Review by vchoi-hdfgroup — APPROVED** [review]

**Pull request merged**